src/ripple/rpc/BookChanges.h

```diff
@@ -88,13 +88,10 @@ computeBookChanges(std::shared_ptr<L const> const& lpAccepted)
                 !node.isFieldPresent(sfPreviousFields))
                 continue;
 
-            STObject& finalFields = (const_cast<STObject&>(node))
-                                        .getField(sfFinalFields)
-                                        .downcast<STObject>();
-
-            STObject& previousFields = (const_cast<STObject&>(node))
-                                           .getField(sfPreviousFields)
-                                           .downcast<STObject>();
+            auto const& ffBase = node.peekAtField(sfFinalFields);
```

Comment:
I don't really care, but I probably would have left off the intermediate `ffBase` and `pfBase` and just done something like:
```
            auto const& finalFields =
                node.peekAtField(sfFinalFields).template downcast<STObject>();
```
It's fine as-is too.

Reply:
That was my initial code and `clang-format` was producing this:
```
            auto const& finalFields =
                node.peekAtField(sfFinalFields).template downcast<STObject>();
            auto const& previousFields = node.peekAtField(sfPreviousFields)
                                             .template downcast<STObject>();
``` 
While this is slightly shorter - it looks so bad that i would probably not sleep tonight if i did this :smiley: 
That's how it ended up being what it is. As the compiler generally does a great job optimizing the intermediate away (generated assembly is the same) I thought it's better be readable than short - I believe there is no harm done :+1: